autoload/LanguageClient.vim

```diff
@@ -1634,4 +1654,82 @@ function! LanguageClient#debugInfo(...) abort
     return LanguageClient#Call('languageClient/debugInfo', l:params, l:Callback)
 endfunction
 
+function! s:ClosePopups(...) abort
+  if s:ShouldUseFloatWindow()
+    call s:CloseFloatingHover()
+  elseif exists('*popup_clear') && s:GetVar('LanguageClient_usePopupHover', v:true)
+    call popup_clear()
+  else
+    :pclose
+  endif
+endfunction
+
+" receives the v:event from the CompleteChanged autocmd
+function! LanguageClient#handleCompleteChanged(event) abort
+  " this timer is just to stop textlock from locking our changes
+  call timer_start(0, funcref('s:ClosePopups'))
+
+  if exists('b:timers') && has_key(b:timers, 'CompleteChanged')
+    call timer_stop(b:timers['CompleteChanged'])
+  else
+    let b:timers = {}
+  endif
```

Comment:
This block here is to add throttling to these calls, as displaying the documentation can be a little expensive, especially for markdown documentation.

Also, and unrelated to this, but maybe we want to take this strategy for `CursorMoved` as well? It would save a lot of unnecessary calls to the server.